packages/gatsby/src/services/run-page-queries.ts

```diff
@@ -0,0 +1,43 @@
+import { processPageQueries } from "../query"
+import { IBuildContext } from "./"
+import reporter from "gatsby-cli/lib/reporter"
+
+export async function runPageQueries({
+  parentSpan,
+  queryIds,
+  store,
+  program,
+  graphqlRunner,
+}: Partial<IBuildContext>): Promise<void> {
+  if (!queryIds || !store) {
```

Comment:
I feel typings are wrong here, these shouldn't be nullable

Reply:
I've made the props object a Partial so that it can have the same type for all services (as it's the context object). Otherwise we'd need to pass loads of unused props into each service when we call it outside the state machine. That means all props are nullable though, so we need the check.